src/app/core/learning-object-module/topics/topics.service.ts

```diff
@@ -45,6 +45,36 @@ export class TopicsService {
     });
   }
 
+  /**
+   *  This function updates the learning the object with the selected tagged topics
+   *
+   * @param username  username of learning object owner
+   * @param id  user id
+   * @param topicIds  id of Topic object
+   */
+  async updateObjectTopics(id: string, topicIds: string[]): Promise<void> {
+    return await new Promise((resolve, reject) => {
+      this.http
+        .patch(TOPICS_ROUTES.UPDATE_TOPIC(id),
+          { topicIds },
+          {
+            headers: this.headers,
+            withCredentials: true,
+            responseType: 'text',
+          }
+        )
+        .pipe(
+
+          catchError(this.handleError)
+        )
+        .toPromise()
+        .then(
+          (res: any) => resolve(res),
+          (err) => reject(err),
+        );
+    });
+  }
```

Comment:
Make sure all the references to the old function get updated to this function.

Reply:
^

Reply:
If you run `npx ng serve` or `ng serve` if you have angular installed. We should be testing the changes we make prior to merging/PR reviews.